[FIX] l10n_fi: adjusting tax report

Megaaaaaa opened the pull request:
Description of the issue this commit addresses:

A new tax has been added and will be used as from the first of september 2024. With this new tax, the tax report has been modified to include it by adding a new line but what's actually required as from the first september is to include the new tax to the existing 24% tax line in the report.

---

Desired behavior after this commit is merged:

The 25.5% tax line in the report includes both 24.0 and 25.5 rates and is named after the 25.5 one but still says that the 24.0 rate taxes are included in it. This will be changed to totally exclude the 24.0 rate once it will not be used anymore probably on the first of january 2025.

---

Documentation:
Change of default VAT rate: https://www.vero.fi/en/businesses-and-corporations/taxes-and-charges/vat/rates-of-vat/new-vat-rate-from-1-september-2024--instructions-for-vat-reporting/
Line about both taxes being in the same line of the report: https://www.vero.fi/en/businesses-and-corporations/taxes-and-charges/vat/rates-of-vat/new-vat-rate-from-1-september-2024--instructions-for-vat-reporting/#:~:text=All%20sales%20with%20both%20the%2024%25%20and%20the%2025.5%25%20rates%20must%20be%20entered%20into%20the%20same%20space%20on%20the%20form

---

task-4101930

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/175963.png)](https://mergebot.odoo.com/odoo/odoo/pull/175963)

casp-odoo reviewed (commented):
Some link to official documentation and maybe, if it exists, a customer ticket mentioning the issue, would be welcomed.
We could also mention the change to come in a comment in account_tax_report_line for posterity. 
Otherwise LGTM @h4818

Megaaaaaa commented:
I added references to the documentation in both the task and the commit message :+1: 
Sadly, this feedback comes from an email discussion with a partner so there is no ticket to reference :smiling_face_with_tear: 
Added some comments as concise as possible, don't hesitate to tell me if you don't like them (I'm not sure I do either :smile:)

h4818 commented on addons/l10n_fi/data/account_tax_template_data.xml:
```diff
@@ -20,7 +20,7 @@
               'repartition_type': 'tax',
               'account_id': ref('account_2930'),
-              'plus_report_line_ids':  [ref('tax_report_sales_25_5')],
+              'plus_report_line_ids':  [ref('tax_report_sales_24')],
```
I don't think these should change

Megaaaaaa replied:
25.5% taxes should never have been put in a separate line in the first place regarding the documentation and the partner's feedback so I stil believe the 25.5% should have been put in the 24% report line.

h4818 replied:
> Using MyTax for VAT reporting
When the VAT taxpayer’s tax period is the month, the VAT return in MyTax will display the following space starting 1 September 2024: “VAT on domestic sales by tax rate: 25.5% VAT”.

IMO, the 25.5% tax tag should stay

Megaaaaaa commented:
> Thanks @Megaaaaaa 🤔 I'm not convinced we should remove it entirely As I understand from the docs provided, you would need to use the 25.5% field when filing online using MyTax. So this should only apply to users who file on paper.
> 
> IMO, we should add a new report line consisting of both 24% and 25.5% taxes. It can be clearly identified as such, and shouldn't impact the totals. That way, when the new forms are printed, we can simply delete that line. WDYT?

Hello :wave:
I still believe we have to entirely remove that line. I don't know if you have this piece of information but that line is a line i added 2-3 weeks ago and realized later that it wasn't the solution legally required thanks o an email from a partner. Businesses in Finland know that the line references both values from 24 and 25.5% even being name 24 before the 1st of September and 25.5 after that date.

Megaaaaaa commented on addons/l10n_fi/data/account_tax_report_line.xml:
```diff
@@ -10,15 +10,10 @@
         <field name="sequence">0</field>
         <field name="report_id" ref="vat_report"/>
     </record>
-    <record id="tax_report_sales_25_5" model="account.tax.report.line">
-        <field name="name">25.5 %:n vero</field>
-        <field name="code">sale_25_5</field>
-        <field name="tag_name">fi_320</field>
-        <field name="sequence">10</field>
-        <field name="report_id" ref="vat_report"/>
-        <field name="parent_id" ref="tax_report_sales_title"/>
-    </record>
```
I don't believe the line will be back soon. There is only one line needed. It used to say 24% and reference 24%. Now it says 24% and refrences 24% and 25.5%. After the first of september until the end of the firscal year, it will say 25.5% and reference 24%. And after that it will say 25.5% and reference 25.5%. So this line won't be back, we'll just rename the 24% to 25.5%.

h4818 commented:
@Megaaaaaa since @Olivier-olma is not available I spoke about this to @chklop 
Since there will be a discrepancy in the electronic vs paper forms, we need to keep both the 25.5% and 24% taxes separate.
Since Businesses in Finland know about this, it will be their responsibility to add the 24% and 25.5% values (for the paper form)
so I guess there is nothing to be done here

Megaaaaaa commented:
> @Megaaaaaa since @Olivier-olma is not available I spoke about this to @chklop Since there will be a discrepancy in the electronic vs paper forms, we need to keep both the 25.5% and 24% taxes separate. Since Businesses in Finland know about this, it will be their responsibility to add the 24% and 25.5% values (for the paper form) so I guess there is nothing to be done here

Okay thank you for investigating it, i'll close the PR :+1:

Megaaaaaa commented:
Brought back to life by a partner's feedback. Specs changed, I'll post a comment when the PR is ready (might wait for a functionnal test once olma is back)

h4818 commented on addons/l10n_fi/data/account_tax_report_line.xml:
```diff
@@ -13,7 +13,7 @@
     <record id="tax_report_sales_25_5" model="account.tax.report.line">
         <field name="name">25.5 %:n vero</field>
```
Spoke to @Olivier-olma, we should include a description something like "(includes 24%)"

h4818 replied:
EDIT:
or "(+ 24 %:n vero)" 
or rather, maybe we should add the comment on the 24% line - that way we won't have to change the 25.5% line in the future?
I leave it up to you

Megaaaaaa replied:
I'd rather not tell what's in the 25.5 line in the 24.0 one so I put "(+ 24 %:n vero)" in the 25.5% line name. Anyway we will have multiple changes to apply on the first of january 2025 (i guess) so changing the line's name in the report is no big deal. Thnak you for asking Olma :+1:

h4818 commented:
@robodoo r+
